### PR TITLE
Use a different Twilio `from` number for pre-prod envs

### LINKED ISSUE #1167: Create separate Twilio phone number for dev/test environments
Have a way to see which messages are sent to prod environments vs. dev/test environments. And which numbers are for TOT vs. Results.

An easy way to do this is create a test number that we use for text messages in non-production environments.

Add separate numbers for different flows- TOT & results.

### PR DESCRIPTION
**Pull request by nullflux**
## Related Issue or Background Info

- Resolves #1167 

## Changes Proposed

- Change the value _default_ to the new pre-prod number
- Explicitly configure the prod number

### CODE REVIEW

File: backend/src/main/java/gov/cdc/usds/simplereport/service/sms/SmsService.java [hunk 1]
```diff
@@ -21,7 +21,7 @@
 public class SmsService {
   private static final Logger LOG = LoggerFactory.getLogger(SmsService.class);
 
-  @Value("${twilio.from-number:+14045312484}")
+  @Value("${twilio.from-number:+12023014570}")
```
Comment (jeskew-gov) [inline backend/src/main/java/gov/cdc/usds/simplereport/service/sms/SmsService.java:24]: Should we set the default `twilio.from-number` in `application.yaml`? Not sure what the best practice here is.

Comment (nullflux) [reply backend/src/main/java/gov/cdc/usds/simplereport/service/sms/SmsService.java:24]: I don't have a preference and I'm also unsure on the best practice 🤷

Comment (nickscialli-usds) [reply backend/src/main/java/gov/cdc/usds/simplereport/service/sms/SmsService.java:24]: I'm a fan of doing it in the application.yml file; keep all the config details out of code. If we needed to switch out the non-prod `from-number`, it'd make a lot more sense to me to look in application.yml than diving into the app code

Comment (nullflux) [reply backend/src/main/java/gov/cdc/usds/simplereport/service/sms/SmsService.java]: Done

**Review by jeskew-gov — APPROVED** [review]

**Pull request merged**